src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -21,27 +21,103 @@ public function __construct(Client $guzzleClient)
     {
-        $request = new Request(
-            'GET',
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              [
+                'Authorization' => sprintf('Bearer %s', $bearer),
+                'Accept' => 'application/json',
+              ],
+              json_encode($bodyParams)
+          );
+        } else {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              ['Authorization' => sprintf('Bearer %s', $bearer)]
+          );
+        }
+
+        return $this->guzzleClient->send($request);
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    protected function deleteApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
+    {
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+            'DELETE',
+            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+            [
+              'Authorization' => sprintf('Bearer %s', $bearer),
+              'Accept' => 'application/json',
+            ],
+            json_encode($bodyParams)
+        );
+        } else {
+            $request = new Request(
+            'DELETE',
             sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
             ['Authorization' => sprintf('Bearer %s', $bearer)]
         );
+        }
 
         return $this->guzzleClient->send($request);
     }
 
     /**
      * @throws GuzzleException
      */
-    protected function postApi(string $uriEndpoint, string $bearer, array $queryParamsMap = []): ResponseInterface
+    protected function postApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
     {
-        $request = new Request(
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+            'POST',
+            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+            [
+              'Authorization' => sprintf('Bearer %s', $bearer),
+              'Accept' => 'application/json',
+            ],
+            json_encode($bodyParams)
+        );
+        } else {
+            $request = new Request(
             'POST',
             sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
             ['Authorization' => sprintf('Bearer %s', $bearer)]
         );
+        }
+
+        return $this->guzzleClient->send($request);
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    protected function putApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
```

Comment:
I know you're adding this method because you're adding DELETE and adding this rounds out the GET/POST/PUT/DELETE set. However, I'm not a fan of adding code that isn't used. This method isn't used anywhere. It can be added later, when it is actually needed. No need to add it now.